plugins/dominant-color-images/hooks.php

```diff
@@ -100,14 +100,18 @@ function dominant_color_img_tag_add_dominant_color( $filtered_image, string $con
 		return $filtered_image;
 	}
 
-	// Only apply the dominant color to images that have a src attribute that
-	// starts with a double quote, ensuring escaped JSON is also excluded.
-	if ( ! str_contains( $filtered_image, ' src="' ) ) {
+	$processor = new WP_HTML_Tag_Processor( $filtered_image );
+	if ( ! $processor->next_tag( array( 'tag_name' => 'IMG' ) ) ) {
+		return $filtered_image;
+	}
+
+	// Only apply the dominant color to images that have a src attribute.
+	if ( null === $processor->get_attribute( 'src' ) || ! is_string( $processor->get_attribute( 'src' ) ) ) {
```

Comment:
This can be further simplified:

```suggestion
	if ( ! is_string( $processor->get_attribute( 'src' ) ) ) {
```